keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1188,239 +937,309 @@ drivers are expected to be written for Kubernetes.
-
-In the Go types above, all structs starting with `NamedResources` are part of
-that structured model. Code generators (more specifically, the applyconfig
-generator) assume that all Go types of an API are defined in the same Go
-package. If it wasn't for that, defining those structs in their own package
-without the `NamedResources` prefix would be possible and make the Go code
-cleaner without affecting the Kubernetes API.
-
-##### ResourceClass
+##### DeviceClass
 
 ```go
-// ResourceClass is used by administrators to influence how resources
-// are allocated.
-//
-// This is an alpha type and requires enabling the DynamicResourceAllocation
-// feature gate.
-type ResourceClass struct {
+// DeviceClass is a vendor or admin-provided resource that contains
+// device configuration and requirements. It can be referenced in
+// the device requests of a claim to apply these presets.
+// Cluster scoped.
+type DeviceClass struct {
     metav1.TypeMeta
     // Standard object metadata
     // +optional
     metav1.ObjectMeta
 
-    // ParametersRef references an arbitrary separate object that may hold
-    // parameters that will be used by the driver when allocating a
-    // resource that uses this class. A dynamic resource driver can
-    // distinguish between parameters stored here and and those stored in
-    // ResourceClaimSpec.
-    // +optional
-    ParametersRef *ResourceClassParametersReference
-
-    // Only nodes matching the selector will be considered by the scheduler
-    // when trying to find a Node that fits a Pod when that Pod uses
-    // a ResourceClaim that has not been allocated yet.
+    // Config defines configuration parameters that apply to each device which is claimed via this class.
+    // Some classses may potentially be satisfied by multiple drivers, so each instance of a vendor
+    // configuration applies to exactly one driver.
+    //
+    // They are passed to the driver, but are not consider while allocating the claim.
     //
-    // Setting this field is optional. If null, all nodes are candidates.
     // +optional
-    SuitableNodes *core.NodeSelector
+    // +listType=atomic
+    Config []ConfigurationParameters
 
-    // DefaultClaimParametersRef is an optional reference to an object that holds parameters
-    // used as default when allocating a claim which references this class. This field is utilized
-    // only when the ParametersRef of the claim is nil. If both ParametersRef
-    // and DefaultClaimParametersRef are nil, the claim requests no resources and thus
-    // can always be allocated.
+    // Requirements must be satisfied by devices. Applies to all devices of
+    // a claim when the claim references the class and only to the devices
+    // in a request when referenced there.
+    //
     // +optional
-    DefaultClaimParametersRef *ResourceClassParametersReference
+    // +listType=atomic
+    Requirements []Requirement
```

Comment:
For aesthetic purposes, I would list the requirements before the config in this struct.

Reply:
Okay.

Reply:
Changed the order and using "Selectors" now.